src/MapMaker.js

```diff
@@ -0,0 +1,60 @@
+const { MAP } = require("./view/stringsUI");
+
+class MapMaker {
+  #upperBridge;
+
+  #lowerBridge;
+
+  constructor(playerArr) {
+    this.#upperBridge = [];
+    this.#lowerBridge = [];
+    this.createMap(playerArr);
+  }
+
+  createMap(playerArr) {
+    playerArr.forEach((playerAction, index) => {
+      this.divideBridge(index);
+      this.handlePlayerAction(playerAction);
+    });
+    this.wrapBridge();
+    return this.getBridge();
+  }
+
+  handlePlayerAction({ selectedMove, isCrossBridge }) {
+    if (selectedMove === MAP.UPPER) {
+      this.#upperBridge.push(MAP[isCrossBridge]);
+      this.#lowerBridge.push(MAP.NONE);
+    } else {
+      this.#lowerBridge.push(MAP[isCrossBridge]);
+      this.#upperBridge.push(MAP.NONE);
+    }
+  }
+
+  divideBridge(index) {
+    if (index > 0) {
+      this.#upperBridge.push(MAP.DIVIDER);
+      this.#lowerBridge.push(MAP.DIVIDER);
+    }
+  }
+
+  wrapBridge() {
+    this.#upperBridge.unshift(MAP.WRAPPER_LEFT);
+    this.#upperBridge.push(MAP.WRAPPER_RIGHT);
+    this.#lowerBridge.unshift(MAP.WRAPPER_LEFT);
+    this.#lowerBridge.push(MAP.WRAPPER_RIGHT);
+  }
+
+  getBridge() {
+    return {
+      upperBridge: this.#upperBridge,
+      lowerBridge: this.#lowerBridge,
+    };
+  }
```

Comment:
wrapBridge 메서드에서 unshift를 사용하고 있는데, push는 O(1)인데 비해 unshift는 O(n)이기 때문에 시간이 오래 걸리는 것 같아 이런 식으로 고쳐보았습니다!

createMap에서 wrapBridge 메서드를 거쳐 getBridge 메서드를 호출하고 있는데, 이렇게 하면 wrapBridge 메서드 없이 getBridge 메서드만으로 해결할 수 있을 것 같아요!😎
```suggestion
  getBridge() {
    return {
      upperBridge: ['[', ...this.#upperBridge, ']'],
      lowerBridge: ['[', ...this.#lowerBridge, ']'],
    };
  }
```

Reply:
... operator를 사용해서 객체 내부 값만 가져는 방법도 있었네요. 리뷰해주셔서 감사합니다! 😃